spectec/spec/wasm-3.0/1-syntax.watsup

```diff
@@ -262,7 +320,8 @@ var mo : memop
 ;; TODO: do c(numtype)?
 syntax c = nat
 syntax c_numtype = nat  ;; TODO
-syntax c_vectype = nat  ;; TODO
+syntax c_vectype = byte*;; TODO
```

Comment:
Our first attempt of using `nat` for `vec_type` was not successful (mainly due to the fact that in algorithmic backends, nat is encoded to `Int64` in Ocaml, which can not store 128 bit numbers), and using `byte*` was a temporary hack (thus TODO).

Considering that we ultimately want to use `i128`, we will try again to see if using nat would be possible.

Reply:
Oh, I see, that's a problem all right. :) Since i128 will be a refinement of nat, that will remain so. We should probably change the implementation of nat and int to use the zarith library (https://github.com/ocaml/Zarith).

Reply:
It turns out that keeping `c_vectype` as `nat` in the spec itself was not that difficult, after all.

By using a bit of hack during translation, algorithmic backends still use bytes sequence (string) as underlying representation for 128bit values, just as reference interpreter does.